tests/ssl-opt.sh

```diff
@@ -13876,15 +13882,124 @@ run_test    "TLS 1.2 ClientHello indicating support for deflate compression meth
+            -s "Consume: waiting for more handshake fragments 256/[0-9]\\+" \
+
+requires_openssl_3_x
+requires_protocol_version tls12
+requires_certificate_authentication
+requires_config_enabled MBEDTLS_SSL_RENEGOTIATION
+run_test    "Handshake defragmentation with client-initiated renegotation: len=512" \
+            "$P_SRV debug_level=4 exchanges=2 renegotiation=1 auth_mode=required" \
+            "$O_NEXT_CLI_RENEGOTIATE -tls1_2 -split_send_frag 512 -cert $DATA_FILES_PATH/server5.crt -key $DATA_FILES_PATH/server5.key -connect 127.0.0.1:+$SRV_PORT" \
+            0 \
+            -s "received TLS_EMPTY_RENEGOTIATION_INFO" \
+            -s "found renegotiation extension" \
+            -s "server hello, secure renegotiation extension" \
+            -s "=> renegotiate" \
+            -S "write hello request" \
+            -s "reassembled record" \
+            -s "initial handshake fragment: 512, 0..512 of [0-9]\\+" \
+            -s "Prepare: waiting for more handshake fragments 512/[0-9]\\+" \
+            -s "Consume: waiting for more handshake fragments 512/[0-9]\\+" \
+
+# Test Server initialised renegotiation with fragmented handshake on TLS1.2
+requires_openssl_3_x
+requires_protocol_version tls12
+requires_certificate_authentication
+requires_config_enabled MBEDTLS_SSL_RENEGOTIATION
+run_test    "Handshake defragmentation with server-initiated renegotation: len=290" \
+            "$O_NEXT_SRV -tls1_2 -split_send_frag 300 -legacy_renegotiation -cert $DATA_FILES_PATH/server5.crt -key $DATA_FILES_PATH/server5.key" \
+            "$P_CLI debug_level=3 renegotiation=1 request_page=/reneg" \
+            0 \
+            -c "initial handshake fragment: 300, 0..300 of [0-9]\\+" \
+            -c "Prepare: waiting for more handshake fragments 300/[0-9]\\+" \
+            -c "Consume: waiting for more handshake fragments 300/[0-9]\\+" \
+            -c "client hello, adding renegotiation extension" \
+            -c "found renegotiation extension" \
+            -c "=> renegotiate"
+
+requires_openssl_3_x
+requires_protocol_version tls12
+requires_certificate_authentication
+requires_config_enabled MBEDTLS_SSL_RENEGOTIATION
+run_test    "Handshake defragmentation with server-initiated renegotation: len=512" \
+            "$O_NEXT_SRV -tls1_2 -split_send_frag 512 -legacy_renegotiation -cert $DATA_FILES_PATH/server5.crt -key $DATA_FILES_PATH/server5.key" \
+            "$P_CLI debug_level=3 renegotiation=1 request_page=/reneg" \
+            0 \
+            -c "initial handshake fragment: 512, 0..512 of [0-9]\\+" \
+            -c "Prepare: waiting for more handshake fragments 512/[0-9]\\+" \
+            -c "Consume: waiting for more handshake fragments 512/[0-9]\\+" \
+            -c "client hello, adding renegotiation extension" \
+            -c "found renegotiation extension" \
+            -c "=> renegotiate"
+
+# Mock negative test to demonstrate the failure with n-bit sized fragments, where ClientHello < n.
```

Comment:
Do you mean n < size of ClientHello?

I'm not sure I understand what you mean by "mock" here.

Reply:
Before [patching ](https://github.com/Mbed-TLS/mbedtls/pull/10007/commits/9796bcd8c2d9f98be39fd70bba28702ed561d993)our client to add the `renegotiation delay` configuration parameter a lot of tests would fail like this one.

Then after fixing it I [have added a new](https://github.com/Mbed-TLS/mbedtls/pull/10007/commits/8158b39f88bf3566d391053eab4102de379a65b0) test to showcase that the issue is actually fixed by the introduction of this parameter 

The intention was to remove those two tests, in the ready to merge code, but left them in for the current review as you many want something similar as part of the coverage